quinn-proto/src/connection/mtud.rs

```diff
@@ -349,6 +348,127 @@ impl SearchState {
     }
 }
 
+#[derive(Clone)]
+struct BlackHoleDetector {
+    /// Counts suspicious packet loss bursts since a packet with size equal to the current MTU was
+    /// acknowledged (or since a black hole was detected)
+    ///
+    /// A packet loss burst is a group of contiguous packets that are deemed lost at the same time
+    /// (see usages of [`MtuDiscovery::on_non_probe_lost`] for details on how loss detection is
+    /// implemented)
+    ///
+    /// A packet loss burst is considered suspicious when it contains only suspicious packets and no
+    /// MTU-sized packet has been acknowledged since the group's packets were sent
+    suspicious_loss_bursts: u8,
+    /// Indicates whether the current loss burst has any non-suspicious packets
+    ///
+    /// Non-suspicious packets are non-probe packets of size <= `BASE_UDP_PAYLOAD_SIZE`
+    loss_burst_has_non_suspicious_packets: bool,
+    /// The largest suspicious packet that was lost in the current burst
+    ///
+    /// Suspicious packets are non-probe packets of size > `BASE_UDP_PAYLOAD_SIZE`
+    largest_suspicious_packet_lost: Option<u64>,
+    /// The largest non-probe packet that was lost (used to keep track of loss bursts)
+    largest_non_probe_lost: Option<u64>,
+    /// The largest acked packet of size `current_mtu`
+    largest_acked_mtu_sized_packet: Option<u64>,
+}
+
+impl BlackHoleDetector {
+    fn new() -> Self {
+        Self {
+            suspicious_loss_bursts: 0,
+            largest_non_probe_lost: None,
+            loss_burst_has_non_suspicious_packets: false,
+            largest_suspicious_packet_lost: None,
+            largest_acked_mtu_sized_packet: None,
+        }
+    }
```

Comment:
Can we ditch this in favor of a `Default` derive?

Reply:
We can't, because a later commit introduced an additional field for `min_mtu`, provided as an argument to `BlackHoleDetector::new`